PennMobile/Home/Cells/Group Invites/HomeGroupInvitesCell.swift

```diff
@@ -0,0 +1,213 @@
+//
+//  HomeGroupInvitesCell.swift
+//  PennMobile
+//
+//  Created by Daniel Salib on 2/7/20.
+//  Copyright © 2020 PennLabs. All rights reserved.
+//
+
+import Foundation
+
+protocol GSRInviteSelectable {
+    func handleInviteSelected(_ invite: GSRGroupInvite, _ accept: Bool)
+}
+
+final class HomeGroupInvitesCell: UITableViewCell, HomeCellConformable {
+    var cardView: UIView! = UIView()
+    
+    static var identifier: String {
+        return "invitesCell"
+    }
+    
+    static func getCellHeight(for item: ModularTableViewItem) -> CGFloat {
+        guard let item = item as? HomeGroupInvitesCellItem else { return 0.0 }
+        
+        // cell height = (invites * inviteHeight) + header + footer + cellInset
+        return (CGFloat(item.invites.count) * GSRGroupInviteCell.cellHeight) + (90.0 + 14.0 + 20.0)
+    }
+    
+    var item: ModularTableViewItem! {
+        didSet {
+            guard let item = item as? HomeGroupInvitesCellItem else { return }
+            setupCell(with: item)
+        }
+    }
+    
+    var invites: GSRGroupInvites?
+    
+    var delegate: ModularTableViewCellDelegate!
+    
+    //var invitesDelegate: HomeViewController
+    
+    fileprivate let safeInsetValue: CGFloat = 14
+    fileprivate var safeArea: UIView!
+    
+    fileprivate var secondaryTitleLabel: UILabel!
+    fileprivate var primaryTitleLabel: UILabel!
+    
+    fileprivate var dividerLine: UIView!
+    fileprivate var groupInvitesTableView: UITableView!
+    
+    // Mark: - Init
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        
+        prepareHomeCell()
+        prepareUI()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Setup UI Elements
+extension HomeGroupInvitesCell {
+    fileprivate func setupCell(with item: HomeGroupInvitesCellItem) {
+        invites = item.invites
+        groupInvitesTableView.reloadData()
+        secondaryTitleLabel.text = "GSR GROUPS"
+        primaryTitleLabel.text = "Pending Invites"
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension HomeGroupInvitesCell: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return invites?.count ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: GSRGroupInviteCell.identifier, for: indexPath) as! GSRGroupInviteCell
+        cell.selectionStyle = UITableViewCell.SelectionStyle.none
+        let invite = invites![indexPath.row]
+        cell.invite = invite
```

Comment:
Bang operators like this make me nervous. Is invites guaranteed to never be `nil`? If so, let's be consistent and make the `invites` field an Implicit Optional (`GSRGroupInvites!` instead of `GSRGroupInvites?`).